### PR TITLE
enh(security): Using the new algorithm to hide the password in the se…

### PR DESCRIPTION
**Pull request by lpinsivy**
…rvice command line

## Description

Based on PR #8781 but adapted for legacy.

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 19.10.x
- [ ] 20.04.x
- [ ] 20.10.x
- [ ] 21.04.x (master)

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/monitoring/objectDetails/common-func.php [hunk 1]
```diff
@@ -38,49 +38,135 @@
...
+            $monitoringCommand,
+            $onDemandServiceMacro,
+            $onDemandHostMacro
+        );
+
+        if (count($monitoringToken) === count($configurationToken)) {
+            $patternMacrosPassword = implode('|', array_keys($macrosPassword));
+            $patternMacrosPassword = str_replace(['$', '~'], ['\$', '\~'], $patternMacrosPassword);
+            foreach ($configurationToken as $index => $token) {
+                if (
```
[the remark this answers is not in the document]
Comment (kduret) [reply www/include/monitoring/objectDetails/common-func.php:128]: can be ignored

**Review by jeremyjaouen — APPROVED** [review]

File: www/include/monitoring/objectDetails/common-func.php [hunk 2]
```diff
@@ -122,48 +222,109 @@ function hidePasswordInCommand($commandName, $hostId, $serviceId)
...
+    $arrtSvcTpl = getListTemplates($pearDB, $serviceId);
+    $arrSvcTplID = array($serviceId);
+    foreach ($arrtSvcTpl as $svc) {
+        $arrSvcTplID[] = $svc['service_id'];
+    }
+
+    // Get list of custom password type macros from service and linked templates
+    $query = "SELECT svc_macro_name, svc_macro_value "
+        . "FROM on_demand_macro_service "
+        . "WHERE is_password = 0 "
```
Comment (sc979) [inline www/include/monitoring/objectDetails/common-func.php:261]: shouldn't it be is_password = 1 ?

Comment (lpinsivy) [reply www/include/monitoring/objectDetails/common-func.php:261]: no this is done by getServiceMacrosPassword function

Comment (sc979) [reply www/include/monitoring/objectDetails/common-func.php:261]: ok

**Review by sc979 — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Review by loiclau — APPROVED** [review]

**Pull request merged**